src/Parser/OrientationParser.js

```diff
@@ -0,0 +1,110 @@
+import * as THREE from 'three';
+
+const DEG2RAD = THREE.Math.DEG2RAD;
+
+// The transform from world to local is  RotationZ(heading).RotationX(pitch).RotationY(roll)
+// The transform from local to world is (RotationZ(heading).RotationX(pitch).RotationY(roll)).transpose()
+function setFromRollPitchHeading(target = new THREE.Quaternion(), roll = 0, pitch = 0, heading = 0) {
+    roll *= DEG2RAD;
+    pitch *= DEG2RAD;
+    heading *= DEG2RAD;
+    // return this.setFromEuler(new THREE.Euler(pitch, roll, heading , 'ZXY')).conjugate();
+    return target.setFromEuler(new THREE.Euler(-pitch, -roll, -heading, 'YXZ')); // optimized version of above
+}
+
+// From DocMicMac, the transform from local to world is:
+// RotationX(omega).RotationY(phi).RotationZ(kappa).RotationX(PI)
+// RotationX(PI) = Scale(1, -1, -1) converts between the 2 conventions for the camera local frame:
+//  X right, Y bottom, Z front : convention in webGL, threejs and computer vision
+//  X right, Y top,    Z back  : convention in photogrammetry
+function setFromOmegaPhiKappa(target = new THREE.Quaternion(), omega = 0, phi = 0, kappa = 0) {
+    omega *= DEG2RAD;
+    phi *= DEG2RAD;
+    kappa *= DEG2RAD;
+    target.setFromEuler(new THREE.Euler(omega, phi, kappa, 'XYZ'));
+    // this.setFromRotationMatrix(new THREE.Matrix4().makeRotationFromQuaternion(this).scale(new THREE.Vector3(1, -1, -1)));
+    target.set(target.w, target.z, -target.y, -target.x); // optimized version of above
+    return target;
+}
+
+// Set East North Up Orientation from geodesic normal
+// target - the quaternion to set
+// up - the normalized geodetic normal to the ellipsoid (given by Coordinates.geodeticNormal)
+var setENUFromGeodesicNormal = (() => {
+    const matrix = new THREE.Matrix4();
+    const elements = matrix.elements;
+    const north = new THREE.Vector3();
+    const east = new THREE.Vector3();
+    return function setENUFromGeodesicNormal(target = new THREE.Quaternion(), up) {
+        // this is an optimized version of matrix.lookAt(up, new THREE.Vector3(0, 0, 0), new THREE.Vector3(0, 0, 1));
+        east.set(-up.y, up.x, 0);
+        east.normalize();
+        north.crossVectors(up, east);
+        north.normalize();
```

Comment:
as `up` and `east` are normalized, `north.normalize()` isn't necessary

Reply:
ok